### PR TITLE
EVG-13151 Add task annotation settings to the build baron admin settings 

### PR DESCRIPTION
**Pull request by malikchaya2**

### CODE REVIEW

File: environment.go [hunk 1]
```diff
@@ -774,9 +774,17 @@ func (e *envState) ClientConfig() *ClientConfig {
 }
 
 type BuildBaronProject struct {
+	// todo: after migrating the data, switch the logic to use this boolean and not the list
+
+	// since we are now also using this struct for task annotations (including the TicketCreateProject field),
+	// we can no longer assume that the presence of a project in the list means that the user wants to run the build baron.
+	showBuildBaron bool `mapstructure:"show_build_baron" bson:"show_build_baron"`
```
Comment (ybrill) [inline environment.go:781]: I don't understand what we would do if this is set to true. I thought once task annotations goes live we will stop displaying the build baron tab.

Comment (ybrill) [inline environment.go:781]: I think the field might need to be public (capitalized) for marshaling/unmarshaling to work. I'm not certain, though.

File: environment.go [hunk 2]
```diff
@@ -785,6 +793,13 @@ type BuildBaronProject struct {
 	BFSuggestionFeaturesURL string `mapstructure:"bf_suggestion_features_url" bson:"bf_suggestion_features_url"`
 }
 
+type AnnotationsSettings struct {
+	// a list of jira fields the user wants to display in addition to state assignee and priority
+	JiraCustomFields []string `mapstructure:"jira_custom_fields" bson:"jira_custom_fields"`
```
Comment (ybrill) [inline environment.go:798]: I think we need a mapping of field names to display names. For instance, Jira could call it `custom_field_123`, so we'd need the user to tell us what to call it in the UI.

Alternatively, we could ask Jira what it's called when we search for the issue. It looks like there's an [extra parameter](https://developer.atlassian.com/cloud/jira/platform/rest/v3/intro/#expansion) (`expand=names`) that will tell us what all the fields are called.

File: environment.go [hunk 3]
```diff
@@ -785,6 +793,13 @@ type BuildBaronProject struct {
...
+	// the endpoint that the user would like to send data to when the file ticket button is clicked
+	WebHook string `mapstructure:"web_hook" bson:"web_hook"`
```
Comment (ybrill) [inline environment.go:800]: maybe we should allow users to specify a secret that gets sent with the webhook, like [GitHub does](https://developer.github.com/webhooks/creating/#secret).
For user subscriptions we generate [a secret](https://github.com/evergreen-ci/evergreen/blob/05d7b78d23cd59007ec5378783ab5aa65fbb704b/public/static/js/subscriptions.js#L140).

**Review by ybrill — CHANGES_REQUESTED** [review]

File: environment.go (hunk 1 above)
Comment (malikchaya2) [reply environment.go:781]: The build baron will still be displayed in the case when the build baron is configured and a task annotation has not been sent.

Comment (malikchaya2) [reply environment.go]: removed in favor of using TicketSearchProjects

File: environment.go (hunk 2 above)
Comment (malikchaya2) [reply environment.go]: added a display text field

File: environment.go (hunk 3 above)
Comment (malikchaya2) [reply environment.go]: done

File: environment.go [hunk 4]
```diff
@@ -785,6 +789,25 @@ type BuildBaronProject struct {
...
+type JiraField struct {
+	// the name that jira calls the field
+	Field string `mapstructure:"field" bson:"field"`
+	// the name the user would like to call it in the UI
+	DisplayText string `mapstructure:"display_text" bson:"display_text"`
+}
+
+type WebHook struct {
+	Endpoint string `mapstructure:"endpoint" bson:"endpoint"`
+	secret   string `mapstructure:"secret" bson:"secret"`
```
Comment (ybrill) [inline environment.go:808]: is this lowercase intentional?

Comment (malikchaya2) [reply environment.go:808]: yes, because we need to be very intentional about how we expose it

Comment (ybrill) [reply environment.go:808]: but how will it be read by the resolver in the graphql package to send down to the UI? How will it be stored in the db (assuming the bson marshaler won't be able to read it)? How will the mapstructure decode it when we need to save it?

Comment (malikchaya2) [reply environment.go]: good point :)

**Review by ybrill — APPROVED** [review]

**Pull request merged**